processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryEngine.java

```diff
@@ -19,24 +19,58 @@
 
 package org.apache.druid.query.timeseries;
 
-import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import org.apache.druid.collections.NonBlockingPool;
+import org.apache.druid.collections.ResourceHolder;
+import org.apache.druid.collections.StupidPool;
+import org.apache.druid.guice.annotations.Global;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.granularity.Granularity;
 import org.apache.druid.java.util.common.guava.Sequence;
+import org.apache.druid.java.util.common.guava.Sequences;
+import org.apache.druid.java.util.common.io.Closer;
+import org.apache.druid.query.QueryContexts;
 import org.apache.druid.query.QueryRunnerHelper;
 import org.apache.druid.query.Result;
 import org.apache.druid.query.aggregation.Aggregator;
+import org.apache.druid.query.aggregation.AggregatorAdapters;
 import org.apache.druid.query.aggregation.AggregatorFactory;
 import org.apache.druid.query.filter.Filter;
-import org.apache.druid.segment.Cursor;
+import org.apache.druid.query.vector.VectorCursorGranularizer;
 import org.apache.druid.segment.SegmentMissingException;
 import org.apache.druid.segment.StorageAdapter;
 import org.apache.druid.segment.filter.Filters;
+import org.apache.druid.segment.vector.VectorColumnSelectorFactory;
+import org.apache.druid.segment.vector.VectorCursor;
+import org.joda.time.Interval;
 
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
 
 /**
  */
 public class TimeseriesQueryEngine
 {
+  private final NonBlockingPool<ByteBuffer> bufferPool;
+
+  /**
+   * Constructor for tests. In production, the @Inject constructor is used instead.
+   */
+  public TimeseriesQueryEngine()
```

Comment:
`@VisibleForTesting` ?

Reply:
@gianm I have received a review request, I suspect you missed this and the following comments.

Reply:
Oh, I did miss this first set of comments. I'll take another look.

Reply:
Did you have any further comments, btw, other than these?

Reply:
Nope, looks good overall.

Reply:
@egor-ryashin Thanks! I just pushed up a new commit addressing your comments.